sentry-core/src/client.rs

```diff
@@ -103,13 +103,18 @@ impl Client {
 
         let mut sdk_info = SDK_INFO.clone();
 
-        // NOTE: We do not filter out duplicate integrations based on their
-        // TypeId.
-        let integrations: Vec<_> = options
-            .integrations
-            .iter()
-            .map(|integration| (integration.as_ref().type_id(), integration.clone()))
-            .collect();
+        let mut integrations = vec![];
+        for integration in &options.integrations {
+            let id = integration.as_ref().type_id();
+            match integrations.iter().position(|(iid, _)| *iid == id) {
```

Comment:
This is the point where creating a dedicated `Integrations` type would pay off. Internally, this could use a `BTreeSet` or `BTreeMap`.